src/compute/app_runner.rs

```diff
@@ -0,0 +1,210 @@
+            temp_env::with_vars_unset(env_vars_to_unset, || {
+                assert_eq!(
+                    start(),
+                    2,
+                    "Should return 2 if get_challenge fails due to missing private key"
+                );
+            });
+        });
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_start_run_fails_get_challenge_succeeds_send_exit_cause_api_error() {
+        let mock_server = MockServer::start().await;
+
+        Mock::given(method("POST"))
+            .and(path(format!("/compute/pre/{}/exit", CHAIN_TASK_ID)))
+            .respond_with(ResponseTemplate::new(500))
+            .mount(&mock_server)
+            .await;
+
+        let mock_server_addr_string = mock_server.address().to_string();
+
+        let result_code = tokio::task::spawn_blocking(move || {
+            let env_vars = vec![
+                (ENV_IEXEC_TASK_ID, Some(CHAIN_TASK_ID)),
+                (ENV_SIGN_WORKER_ADDRESS, Some(WORKER_ADDRESS)),
+                (
+                    ENV_SIGN_TEE_CHALLENGE_PRIVATE_KEY,
+                    Some(ENCLAVE_CHALLENGE_PRIVATE_KEY),
+                ),
+                (ENV_WORKER_HOST, Some(mock_server_addr_string.as_str())),
+            ];
+
+            temp_env::with_vars(env_vars, || start())
+        })
+        .await
+        .expect("Blocking task panicked");
+
+        assert_eq!(
+            result_code, 2,
+            "Should return 2 if sending exit cause to worker API fails"
+        );
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_start_run_fails_get_challenge_succeeds_send_exit_cause_api_success() {
+        let mock_server = MockServer::start().await;
+
+        let expected_cause_enum = ReplicateStatusCause::PreComputeFailedUnknownIssue;
+        let expected_exit_message_payload = json!({
+            "cause": expected_cause_enum // Relies on ReplicateStatusCause's Serialize impl
+        });
+
+        // Mock the worker API to return success
+        Mock::given(method("POST"))
+            .and(path(format!("/compute/pre/{}/exit", CHAIN_TASK_ID)))
+            .and(body_json(expected_exit_message_payload))
+            .respond_with(ResponseTemplate::new(200))
+            .mount(&mock_server)
+            .await;
+
+        let mock_server_addr_string = mock_server.address().to_string();
+
+        // Move the blocking operations into spawn_blocking
+        let result_code = tokio::task::spawn_blocking(move || {
+            let env_vars = vec![
+                (ENV_IEXEC_TASK_ID, Some(CHAIN_TASK_ID)),
+                (ENV_SIGN_WORKER_ADDRESS, Some(WORKER_ADDRESS)),
+                (
+                    ENV_SIGN_TEE_CHALLENGE_PRIVATE_KEY,
+                    Some(ENCLAVE_CHALLENGE_PRIVATE_KEY),
+                ),
+                (ENV_WORKER_HOST, Some(mock_server_addr_string.as_str())),
+            ];
+
+            temp_env::with_vars(env_vars, || start())
```

Comment:
Clippy recommandation
```suggestion
            temp_env::with_vars(env_vars, start)
```